lib/std/os.zig

```diff
@@ -5250,6 +5250,32 @@ pub fn poll(fds: []pollfd, timeout: i32) PollError!usize {
     }
 }
 
+pub const PPollError = error{
+    /// The operation was interrupted by a delivery of a signal before it could complete.
+    SignalInterrupt,
```

Comment:
The precedent of all the other functions in this file is to loop and retry on EINTR. You're welcome to make the case for changing this in the std lib but for this PR it should follow the pattern.

Reply:
Yes, this is intentional, since the main point of `ppoll` is that it allows an application to wait until either a file descriptor becomes ready or until a signal is caught. Retrying on EINTR defeats the whole purpose of this function.

Quoting from linux manpages:

> The relationship between **poll**() and **ppoll**() is analogous to the relationship between **select**(2) and **pselect**(2): like **pselect**(2), **ppoll**() allows an application to safely wait until either a file descriptor becomes ready or until a signal is caught.

> The reason that **pselect**() is needed is that if one wants to wait for either a signal or for a file descriptor to become ready, then an atomic test is needed to prevent race conditions. (Suppose the signal handler sets a global flag and returns. Then a test of this global flag followed by a call of **select**() could hang indefinitely if the signal arrived just after the test but just before the call. By contrast, **pselect**() allows one to first block signals, handle the signals that have come in, then call **pselect**() with the desired sigmask, avoiding the race.)